enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/state/RaftState.java

```diff
@@ -47,32 +47,32 @@
     private final Log log;
     private final RaftLog entryLog;
     private final InFlightMap<RaftLogEntry> inFlightMap;
+    private final boolean supportPreVoting;
 
     private TermState termState;
     private VoteState voteState;
 
     private MemberId leader;
     private Set<MemberId> votesForMe = new HashSet<>();
+    private Set<MemberId> preVotesForMe = new HashSet<>();
     private Set<MemberId> heartbeatResponses = new HashSet<>();
     private FollowerStates<MemberId> followerStates = new FollowerStates<>();
     private long leaderCommit = -1;
     private long commitIndex = -1;
     private long lastLogIndexBeforeWeBecameLeader = -1;
+    private boolean isPreElection = true; // Needed to form a cluster
```

Comment:
Why is this needed? Is this for unit tests? Perhaps unit tests should generally run with pre-election support off then? ITs etc should be able to form clusters even with the pre-election protocol extension in effect.

Reply:
This is needed for integration tests. If a cluster member is not in pre-election state when pre-election support is enabled then it won't give a positive pre-vote response to a pre-vote request, and a real election won't occur.

Reply:
Hmm.. I think I still don't get why this needs to be set. The pre-election state should not have to be set to give a positive pre-vote response? And an instance seeking an election should be able to reach the isPreElection status quite quickly right?